nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/xml/XMLRecordSetWriter.java

```diff
@@ -0,0 +1,196 @@
+    public static final AllowableValue NEVER_SUPPRESS = new AllowableValue("never-suppress", "Never Suppress",
+            "Fields that are missing (present in the schema but not in the record), or that have a value of null, will be written out as a null value");
+    public static final AllowableValue SUPPRESS_MISSING = new AllowableValue("suppress-missing", "Suppress Missing Values",
+            "When a field has a value of null, it will be written out. However, if a field is defined in the schema and not present in the record, the field will not be written out.");
+
+    public static final AllowableValue USE_PROPERTY_AS_WRAPPER = new AllowableValue("use-property-as-wrapper", "Use Property as Wrapper",
+            "The value of the property \"Array Tag Name\" will be used as the tag name to wrap elements of an array. The field name of the array field will be used for the tag name " +
+                    "of the elements.");
+    public static final AllowableValue USE_PROPERTY_FOR_ELEMENTS = new AllowableValue("use-property-for-elements", "Use Property for Elements",
+            "The value of the property \"Array Tag Name\" will be used for the tag name of the elements of an array. The field name of the array field will be used as the tag name " +
+                    "to wrap elements.");
+    public static final AllowableValue NO_WRAPPING = new AllowableValue("no-wrapping", "No Wrapping",
+            "The elements of an array will not be wrapped");
+
+    public static final PropertyDescriptor SUPPRESS_NULLS = new PropertyDescriptor.Builder()
+            .name("suppress_nulls")
+            .displayName("Suppress Null Values")
+            .description("Specifies how the writer should handle a null field")
+            .allowableValues(NEVER_SUPPRESS, ALWAYS_SUPPRESS, SUPPRESS_MISSING)
+            .defaultValue(NEVER_SUPPRESS.getValue())
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor PRETTY_PRINT_XML = new PropertyDescriptor.Builder()
+            .name("pretty_print_xml")
+            .displayName("Pretty Print XML")
+            .description("Specifies whether or not the XML should be pretty printed")
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor ROOT_TAG_NAME = new PropertyDescriptor.Builder()
+            .name("root_tag_name")
+            .displayName("Name of Root Tag")
+            .description("Specifies the name of the XML root tag wrapping the record set")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .defaultValue("root")
+            .required(true)
+            .build();
+

```

Comment:
Initially, I planned to use the record name of the schema for this using
{code}
recordSchema.getIdentifier().getName().get()
{code}
but for my test cases (add schema as text / attribute) this was null.

Reply:
ok. shall I create a ticket for this?

Reply:
@JohannesDaniel @markap14 I just merged the linked ticket. You can rebase now.